src/stdlib/base.rs

```diff
@@ -8,6 +8,76 @@ use crate::{
 };
 
 pub fn load_base<'gc>(ctx: Context<'gc>) {
+    ctx.set_global(
+        "tonumber",
+        Callback::from_fn(&ctx, |ctx, _, mut stack| {
+            fn extract_number_data(bytes: &[u8]) -> (&[u8], i64) {
+                let first_nonspace = bytes.iter().position(|b| !b.is_ascii_whitespace());
+                let last_nonspace = bytes.iter().rposition(|b| !b.is_ascii_whitespace());
+                let bytes = match (first_nonspace, last_nonspace) {
+                    (Some(f), Some(l)) => &bytes[f..=l],
+                    (Some(f), None) => &bytes[f..],
+                    (None, Some(l)) => &bytes[..=l],
+                    (None, None) => bytes,
+                };
+
+                let is_neg = bytes.first().is_some_and(|b| *b == b'-');
+                let bytes = if is_neg || bytes.first().is_some_and(|b| *b == b'+') {
+                    &bytes[1..]
+                } else {
+                    bytes
+                };
+                let sign = if is_neg { -1 } else { 1 };
+                (bytes, sign)
```

Comment:
This is the same as `read_neg` from the lexer which can be made public so it can be used here as well.